cirq/work/observable_measurement.py

```diff
@@ -211,3 +269,148 @@ def _subdivide_meas_specs(
         repetitions //= 2
 
     return flippy_mspecs, repetitions
+
+
+def _to_sweep(param_tuples):
+    """Turn param tuples into a sweep."""
+    to_sweep = [dict(pt) for pt in param_tuples]
+    to_sweep = study.to_sweep(to_sweep)
+    return to_sweep
+
+
+def _get_qubits(max_settings: Iterable[InitObsSetting]):
+    """Helper function to find all the qubits in a suite of settings used
+    in `measure_grouped_settings`.
+
+    Qubits are returned in sorted order.
+    """
+    qubits = set()
+    for max_setting in max_settings:
+        max_in_st = max_setting.init_state
+        qubits |= set(max_in_st.qubits)
+    return sorted(qubits)
+
+
+def _determine_needs_init_layer(init_state: value.ProductState):
+    """Helper function to determine whether we need the initial layer of
+    rotations used in `measure_grouped_settings`.
+    """
+    for _, st in init_state:
```

Comment:
it would be more pythonic to just `return any(st is not value.KET_ZERO for _, st in init_state)`.

Reply:
good call. changed and inlined this call; removed helper function.